fortitude/src/check.rs

```diff
@@ -245,7 +246,13 @@ pub(crate) fn check_file(
     }
 
     // Perform AST analysis
-    let tree = parse(file.source_text())?;
+    let mut parser = Parser::new();
+    parser
+        .set_language(&tree_sitter_fortran::LANGUAGE.into())
+        .context("Error loading Fortran grammar")?;
```

Comment:
```suggestion
    let mut parser = Parser::new()
        .set_language(&tree_sitter_fortran::LANGUAGE.into())
        .context("Error loading Fortran grammar")?;
```

Reply:
Unfortunately you can't do this, as `set_language` returns `Result<(), LanguageError>`.